File: NotificationAPI.Infrastructure/Data/ApplicationDbContext.cs
using Microsoft.EntityFrameworkCore;
using NotificationAPI.Domain.Entities;

namespace NotificationAPI.Infrastructure.Data;

public class ApplicationDbContext : DbContext
{
    public DbSet<User> Users { get; set; }
    public DbSet<Notification> Notifications { get; set; }

    public ApplicationDbContext(DbContextOptions<ApplicationDbContext> options) : base(options)
    {
    }

    protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        base.OnModelCreating(modelBuilder);

        modelBuilder.Entity<User>()
            .HasKey(u => u.Id);

        modelBuilder.Entity<User>()
            .Property(u => u.Name)
            .IsRequired()
            .HasMaxLength(100);

        modelBuilder.Entity<User>()
            .Property(u => u.Email)
            .IsRequired()
            .HasMaxLength(255);

        modelBuilder.Entity<User>()
            .HasIndex(u => u.Email)
            .IsUnique();

        modelBuilder.Entity<Notification>()
            .HasKey(n => n.Id);

        modelBuilder.Entity<Notification>()
            .Property(n => n.Title)
            .IsRequired()
            .HasMaxLength(200);

        modelBuilder.Entity<Notification>()
            .Property(n => n.Message)
            .IsRequired()
            .HasMaxLength(1000);

        modelBuilder.Entity<Notification>()
            .HasOne(n => n.User)
            .WithMany()
            .HasForeignKey(n => n.UserId);
    }
}

File: NotificationAPI.Infrastructure/Data/ApplicationDbContext.vb
Imports Microsoft.EntityFrameworkCore
Imports NotificationAPI.Domain.Entities

Namespace NotificationAPI.Infrastructure.Data

    Public Class ApplicationDbContext
        Inherits DbContext

        Public Property Users As DbSet(Of User)
        Public Property Notifications As DbSet(Of Notification)

        Public Sub New(options As DbContextOptions(Of ApplicationDbContext))
            MyBase.New(options)
        End Sub

        Protected Overrides Sub OnModelCreating(modelBuilder As ModelBuilder)
            MyBase.OnModelCreating(modelBuilder)

            ' User entity configuration
            modelBuilder.Entity(Of User)() _
                .HasKey(Function(u) u.Id)

            modelBuilder.Entity(Of User)() _
                .Property(Function(u) u.Name) _
                .IsRequired() _
                .HasMaxLength(100)

            modelBuilder.Entity(Of User)() _
                .Property(Function(u) u.Email) _
                .IsRequired() _
                .HasMaxLength(255)

            modelBuilder.Entity(Of User)() _
                .HasIndex(Function(u) u.Email) _
                .IsUnique()

            ' Notification entity configuration
            modelBuilder.Entity(Of Notification)() _
                .HasKey(Function(n) n.Id)

            modelBuilder.Entity(Of Notification)() _
                .Property(Function(n) n.Title) _
                .IsRequired() _
                .HasMaxLength(200)

            modelBuilder.Entity(Of Notification)() _
                .Property(Function(n) n.Message) _
                .IsRequired() _
                .HasMaxLength(1000)

            ' One-to-Many: User -> Notifications
            modelBuilder.Entity(Of Notification)() _
                .HasOne(Function(n) n.User) _
                .WithMany() _
                .HasForeignKey(Function(n) n.UserId)
        End Sub
    End Class

End Namespace
